### ISSUE TITLE
🐛 source-mssql: connector forces snapshot isolation level

### ISSUE THREAD

**Issue by sh4sh** [airbytehq/airbyte#28543]
### Connector Name

source-mssql

### Connector Version

1.1.0

### What step the error happened?

Configuring a new connector

### Revelant information

PR with proposed fix: https://github.com/airbytehq/airbyte/pull/28545

There appears to be a bug in source-mssql's support for CDC + Read Committed. Currently both snapshot isolation levels check the db for enabled full `Snapshot Isolation`, which it should not do when the `Read Committed` isolation level is selected.
 
Support for non-snapshot CDC is added here: https://github.com/airbytehq/airbyte/pull/13168 
In the connector spec we see a rework of `replication_method` to support the `Read Committed` isolation level.  `replication_method` field is renamed to `replication` for hygiene.
We can see from comments that the new implementation since 0.4.0 should use the `replication` parameter instead of the `replication_method`. More recently however, the connector was modified in https://github.com/airbytehq/airbyte/pull/16215 to use `replication_method` which has caused some problems.

In the original change to support `Read Committed` snapshot isolation level for MSSQL + CDC,  we see the new `REPLICATION_FIELD` value pulled into `CdcMssqlHelper.java`:
https://github.com/airbytehq/airbyte/blob/4284ce6b17d324887f3e51226ee481484854c7fd/airbyte-integrations/connectors/source-mssql/src/main/java/io/airbyte/integrations/source/mssql/MssqlCdcHelper.java#L15-L20

`isCdc` checks whether CDC is enabled. It supports both `LEGACY_REPLICATION_FIELD` and `REPLICATION_FIELD`:
https://github.com/airbytehq/airbyte/blob/4284ce6b17d324887f3e51226ee481484854c7fd/airbyte-integrations/connectors/source-mssql/src/main/java/io/airbyte/integrations/source/mssql/MssqlCdcHelper.java#L95-L107

`getSnapshotIsolationConfig` checks the snapshot isolation level:

https://github.com/airbytehq/airbyte/blob/4284ce6b17d324887f3e51226ee481484854c7fd/airbyte-integrations/connectors/source-mssql/src/main/java/io/airbyte/integrations/source/mssql/MssqlCdcHelper.java#L109-L122

Since then, the spec parameter was modified from `replication` back to `replication_method`.  This means that the `getSnapshotIsolationConfig` function still needs `REPLICATION_FIELD` to be `hasNonNull`. Unfortunately however it's always null, because `REPLICATION_FIELD` doesn't exist anymore, we only have `LEGACY_REPLICATION_FIELD`.

So now, if we look at master branch, we observe that the connector config is not passed into some functions.
i.e.,

https://github.com/airbytehq/airbyte/blob/7e4797d90dc7ddfdef2bc1ba85d6a91550397da2/airbyte-integrations/connectors/source-mssql/src/main/java/io/airbyte/integrations/source/mssql/MssqlCdcHelper.java#L120-L140

tl;dr quick/non-breaking-change fix is to update functions with the correct config parameter

Since this work was originally done to make naming consistent throughout CDC connectors (a reasonable goal) we should probably look at refactoring.

### Contribute

- [X] Yes, I want to contribute
